### PR TITLE
Fix #565: App on-boarding flag

### LINKED ISSUE #565: Onboarding flow need domain layer controller
Onboarding flow should be visible only once irrespective of learner profile. Introduce domain layer code to control this feature.

### PR DESCRIPTION
**Pull request by nikitamarysolomanpvt**
## Explanation
Duplicate of #574
Review part 1 - #618 first.
Onboarding flow should be visible only once irrespective of learner profile. Introduced domain layer code to control this feature.
Oppia app now showing On-boarding on initial app open, and directly to choose profile page upon returning to the app. Both of these states are also properly retained across configuration changes due to Dagger not leading to activities trying to recreate their dependent controllers.

From a testing side, testcase for OnboardingFlowControllerTest is introduced to test OnboardingFlowController.

Also this added new SplashActivity tests which rely on sleeping rather than idling resource similar like HomeActivity.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: domain/src/main/java/org/oppia/domain/OnBoardingFlowController.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.Logger
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Controller for persisting and retrieving the user on-boarding information of the app. */
+@Singleton
+class OnBoardingFlowController @Inject constructor(
```
Comment (veena14cs) [inline domain/src/main/java/org/oppia/domain/OnBoardingFlowController.kt]: This file shouldn't be included here as it is a part of domain layer.

File: domain/src/test/java/org/oppia/domain/OnboardingFlowControllerTest.kt [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+import kotlin.coroutines.EmptyCoroutineContext
+
+/** Tests for [OnBoardingFlowController]. */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class OnBoardingFlowControllerTest {
```
Comment (veena14cs) [inline domain/src/test/java/org/oppia/domain/OnboardingFlowControllerTest.kt]: Ditto

File: model/src/main/proto/example.proto [hunk 3]
```diff
@@ -9,6 +9,10 @@ message UserAppHistory {
   bool already_opened_app = 1;
 }
 
+message OnBoardingFlow {
```
Comment (veena14cs) [inline model/src/main/proto/example.proto]: ditto

**Review by veena14cs — CHANGES_REQUESTED** [review]
Please remove domain code from here as this PR should only focus on App module.

**Review by rt4914 — COMMENTED** [review]
Please address @veena14cs 's comment, else LGTM

File: domain/src/main/java/org/oppia/domain/OnBoardingFlowController.kt (hunk 1 above)
Comment (nikitamarysolomanpvt) [reply domain/src/main/java/org/oppia/domain/OnBoardingFlowController.kt]: This seems to be issue with Git i merged with parent branch and now not showing.

File: domain/src/test/java/org/oppia/domain/OnboardingFlowControllerTest.kt (hunk 2 above)
Comment (nikitamarysolomanpvt) [reply domain/src/test/java/org/oppia/domain/OnboardingFlowControllerTest.kt]: Ditto

File: model/src/main/proto/example.proto (hunk 3 above)
Comment (nikitamarysolomanpvt) [reply model/src/main/proto/example.proto]: Ditto

**Review by veena14cs — APPROVED** [review]
LGTM.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Comment by rt4914** [comment]
@nikitamarysolomanpvt You can close this PR. This has been included in #558

**Comment by nikitamarysolomanpvt** [comment]
Rajat included these files in his on-boarding flow PR.

**Pull request closed without merging**